http/server.go

```diff
@@ -112,3 +114,31 @@ func GetRequestID(r *http.Request) string {
 func GetUserID(r *http.Request) string {
 	return strx.Or(r.Header.Get("X-User"), r.Header.Get("X-User-Id"))
 }
+
+// GetSkipAndCount returns the skip and count pagination values from the query parameters
+// skip is the number of records to pass over before starting a search
+// count is the number of records to retrieve in the search
+// exists indicates if skip or count was passed into the request URL
+func GetSkipAndCount(r *http.Request) (skip int, count int, exists bool, errors []error) {
+	skipVal := r.URL.Query().Get("skip")
+	var err error
+	skip, err = strconv.Atoi(skipVal)
+	skip = int(math.Min(float64(skip), 10000))
+	skip = int(math.Max(0, float64(skip)))
+	if err != nil && skip == 0 && skipVal != "" {
+		errors = append(errors, err)
```

Comment:
I think we should return the first error. If we can't parse skip/count then we should likely fail the request.

Reply:
I updated it to append the err only if they passed in a value so we won't return errors if they didn't pass in anything.

Reply:
I don't understand why we're returning `[]error` though. Can we just return the first `error`?

Reply:
Yeah, that was confusing. I was trying to be too informative. Refactored to just return an err.